Hantar/code.js

```diff
@@ -0,0 +1,403 @@
+var LS_PREFIX = "hantar_";
+var LS_KEY_SESSION_ID = LS_PREFIX + "session_id";
+var LS_KEY_UNMARKED_LIST = LS_PREFIX + "unmarked_list";
+var LS_KEY_MARKED_LIST = LS_PREFIX + "marked_list";
+
+var LIST_SIGNATURE_REGEX = /^(?:(?:^|,)(\d+(-\d+)?))+$/g;
+
+var ELM_ID_LISTS = ["unmarked-list", "marked-list"];
+
+var ELM_ID_MAIN_MENU = "main-menu";
+var ELM_ID_MAIN_SCREEN = "main-screen";
+
+var ELM_ID_INPUT = "num-input";
+var ELM_ID_MARK_BTN = "mark-btn";
+
+var ELM_ID_QUICK_FIX_CHKBX = "allow-click-chkbx";
+
+var LIST_ITEM_TAG = "li";
+
+var MARK_BTN_ONCLICK_SINGLE = "mark_num(this.previousElementSibling.value);this.previousElementSibling.value=''"
+var MARK_BTN_ONCLICK_MULTIPLE = "mark_nums(this.previousElementSibling.value);this.previousElementSibling.value=''"
+
+var INPUT_TYPE_MODE_SINGLE = "tel";
+var INPUT_TYPE_MODE_MULTIPLE = "text";
+
+var INPUT_MODE_SWITCH = {
+    "single": {"type": INPUT_TYPE_MODE_SINGLE, "onclick": MARK_BTN_ONCLICK_SINGLE},
+    "multiple": {"type": INPUT_TYPE_MODE_MULTIPLE, "onclick": MARK_BTN_ONCLICK_MULTIPLE}
+};
+
+init();
+
+function log(str) { console.log(str); }
+
+function load(key)
+{
+    return localStorage.getItem(key);
+}
+
+function save(key, value)
+{
+    localStorage.setItem(key, value);
+}
+
+function range(start, end, step=1)
+{
+    var list = []
+    var max = Math.max(start, end);
+    var min = Math.min(start, end);
+    for (var i = min; i <= max; i += step)
+    {
+        list.push(i);
+    }
+    return list;
+}
+
+function sort(arr, remove_dups=false)
+{
+    arr.sort(function(a, b) { return (a - b); });
+    
+    if (remove_dups)
+    {
+        arr = arr.filter(function(item, index) { return arr.indexOf(item) == index });
+    }
+    
+    return arr;
+}
+
+function gen_num_list(list_signature)
+{
+    if (list_signature == "")
+    {
+        return [];
+    }
+    
+    var list = [];
+    var items = list_signature.split(",");
+    var range_parts;
+    for (var i = 0, len = items.length; i < len; i++)
+    {
+        if (items[i].indexOf("-") == -1)
+        {
+            list.push(items[i] * 1);
+        }
+        else
+        {
+            range_parts = items[i].split("-");
+            list = list.concat(range(range_parts[0] * 1, range_parts[1] * 1));
+        }
+    }
+    return list;
+}
+
+function gen_list_signature(num_list)
```

Comment:
Very complex, try simplifying it. Is there any simpler way to save the information you need?

Reply:
Try looking for a format that you won't have to parse all by yourself.